lib/src/Uncertainty/Distribution/UniformOverMesh.cxx

```diff
@@ -194,33 +194,32 @@ Scalar UniformOverMesh::computePDF(const Point & point) const
 }
 
 
-/* Get the CDF of the distribution, i.e. P(X <= point) = CDF(point) */
-Scalar UniformOverMesh::computeCDF(const Point & point) const
+/* Get the probability content of an interval */
+Scalar UniformOverMesh::computeProbabilityContinuous(const Interval & interval) const
 {
-  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
-  Scalar cdf = 0.0;
-  const Interval quadrant(Point(getDimension(), -SpecFunc::Infinity), point);
-  const Interval intersection(quadrant.intersect(getRange()));
+  if (interval.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "UniformOverMesh interval must have dimension " << getDimension() << ", got " << interval.getDimension();
+  Scalar probability = 0.0;
+  const Interval intersection(interval.intersect(getRange()));
   if (intersection.isEmpty())
-    cdf = 0.0;
+    probability = 0.0;
   else if (intersection == getRange())
```

Comment:
This test will fail for distributions with unbounded range. The variable quadrant has been built using two points, so its bounds are finite. As a result intersection has finite bounds, and if range has infinite bounds the test will fail even if the numerical bounds are equal. Either quadrant is built using the finite flag of the lower bound, or the test is based on bounds only (my preference):
```
else if (intersection.getLowerBound() == getRange().getLowerBound() && intersection.getUpperBound() == getRange().getUpperBound())
```

Reply:
assuming you're referring to TruncatedOverMesh, this is ok since the range is also computed as an intersection with the mesh range, so a finite interval too, I just checked with the Normal from the test
for UniformOverMesh the range is just the mesh range of course